addons/hr/models/hr_department.py

```diff
@@ -53,6 +53,32 @@ def _search_has_read_access(self, operator, value):
         departments_ids = self.env['hr.department'].sudo().search([('manager_id', 'in', self.env.user.employee_ids.ids)]).ids
         return [('id', 'child_of', departments_ids)]
 
+    def _search_complete_name(self, operator, value):
+        supported_operators = ["=", "!=", "ilike", "not ilike", "in", "not in", "=ilike"]
+        if operator not in supported_operators or not isinstance(value, (str, list)):
+            raise NotImplementedError(_('Operation not Supported.'))
+        department = self.env['hr.department'].search([])
+        if operator == '=':
+            department = department.filtered(lambda m: m.complete_name == value)
+        elif operator == '!=':
+            department = department.filtered(lambda m: m.complete_name != value)
+        elif operator == 'ilike':
+            department = department.filtered(lambda m: value.lower() in m.complete_name.lower())
+        elif operator == 'not ilike':
+            department = department.filtered(lambda m: value.lower() not in m.complete_name.lower())
+        elif operator == 'in':
+            department = department.filtered(lambda m: m.complete_name in value)
+        elif operator == 'not in':
+            department = department.filtered(lambda m: m.complete_name not in value)
+        elif operator == '=ilike':
+            if value.startswith('%'):
```

Comment:
???

Reply:
@xmo-odoo What do you think ?

Reply:
The semantics are incorrect. % is a multi-character wildcard; you can have it anywhere in the value. And it also ignores single-character wildcards.

The correct way, I think, is something like p = re.compile(re.escape(value).replace('%', '.*').replace('_', '.'), flags=re.IGNORECASE), and then p.fullmatch(m.complete_name).